alacritty_terminal/src/event_loop.rs

```diff
@@ -261,13 +261,10 @@ where
                                 if let Some(code) = code {
                                     self.event_proxy.send_event(Event::ChildExit(code));
                                 }
-                                if self.hold {
-                                    // With hold enabled, make sure the PTY is drained.
+                                if self.drain_on_exit {
                                     let _ = self.pty_read(&mut state, &mut buf, pipe.as_mut());
-                                } else {
-                                    // Without hold, shutdown the terminal.
-                                    self.terminal.lock().exit();
                                 }
+                                self.terminal.lock().exit();
```

Comment:
@chrisduerr do you think it would be better to just send exit and not lock the terimnal while at it, I really don't get why we lock it here, seems pointless given that we just send event via _proxy_.

Reply:
Yeah generally speaking it shouldn't be necessary. That said, it shouldn't make any difference for this specific operation anyway.